jerry-core/debugger/debugger.h

```diff
@@ -0,0 +1,39 @@
+/* Copyright 2016 University of Szeged.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef JERRY_DEBUGGER_H
+#define JERRY_DEBUGGER_H
+
+/**
+ *  Package types
+ */
+typedef enum 
+{
+  SOURCE_FILE,                /**< name of the source file */
+  FUNCTION_NAME,              /**< name of the parsed function */
+  BREAKPOINT_INFO,            /**< breakpoint */
+} package_type_t;
+
+typedef struct 
+{
+  package_type_t type;        /**< type of the package */
+  size_t msg_len;             /**< the lenght of the literal array */
+  char msg[128];              /**< the literal array */
+  uint32_t  line_offset;      /**< line offset */
+  uint32_t  line_index;       /**< line index */
+  uint16_t count;             /**< line count */
```

Comment:
There should not be packed everything into one structure. The layout should look like this (without comments):

    typdef struct
    {
      uint8_t type;
      uint8_t size;
    } jerry_debug_message_header;

And each message should have its own type.

    typdef struct
    {
      jerry_debug_message_header header;
      char file_name[1];
    } jerry_debug_message_source_name;

One structure for each message.

Perhaps SOURCE_FILE type is enough for this patch.